ui/views.py

```diff
@@ -494,3 +497,40 @@ def get(self, request, *args, **kwargs):
             else:
                 return redirect('/')
         return super().get(request, *args, **kwargs)
+
+
+class MoiraListsForUser(APIView):
+    """
+    View for getting moira lists against given user.
+    """
+    authentication_classes = (authentication.SessionAuthentication,)
+    permission_classes = (permissions.IsAdminUser,)
+
+    def get(self, request, username_or_email):
+        """Get and return the list names"""
+        regex = r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"
+
+        email = username_or_email
+        if not username_or_email.endswith("@mit.edu"):
```

Comment:
This will break when supplied argument is a valid email address but not ends with `@mit.edu`, e.g `tasawernawaz@gmail.com`

Reply:
This seems to work with all emails:
```python
    def get(self, request, username_or_email):
        """Get and return the list names"""

        email = username_or_email
        if '@' not in email:
            email = "{username}@mit.edu".format(username=username_or_email)

        try:
            user = User.objects.get(Q(username=username_or_email) | Q(email=email))
        except User.DoesNotExist:
            return Response(status.HTTP_404_NOT_FOUND)

        user_lists = query_moira_lists(user)
        return Response(data={'user_lists': user_lists})
```

but you can keep it this way if you prefer.  However in that case it might be best to make use of django's built in email validator if you stick with this approach, rather than a custom regex (`from django.core.validators import validate_email`)